### ISSUE TITLE
Monitor Version 2.0.2 not running

### ISSUE THREAD

**Issue by dwlahnum** [Legedric/ptmagic#549]
My Start PTMagic Monitor.cmd is showing that everything is working fine, but I can't launch the
 monitor in my browser. Is there a new way other than the IP or
 http://localhost:5000 now? I'm currently running PTMagic version 2.0.2 with the dotnet 2.1.0, I know the pic shows the path as being 1.4.0, but it's not that version.
 [image: image]
 <https://user-images.githubusercontent.com/39037555/40977475-14e7a670-689f-11e8-8dbc-32e98023c919.png>

**Comment by Legedric** [comment]
No, nothing changed in that regard. If you cannot open the monitor then you did something wrong. Unfortunately it is extremely hard for me to find out what as I do not have this issue on my machines here so I am not able to reproduce it.

Make sure that:
- You installed the correct .NET Core package (2.1)
- Just to be sure: Uninstall all other .NET Core packages
- Nothing of your security system (for example) fire wall blocks access to this website

**Comment by Tomotoman** [comment]
I'm having the same problem. No errors in the CMD window.
I installed the  .NET Core package (2.1.0 RUNTIME) and uninstalled 2.0.7
Security system wasn't blocking it before, so i'm sure that's not the problem.

**Comment by LeeUmney** [comment]
I get a 500 error when trying to open the PT Magic webpage, all pre-reqs met and firewall opened for port 5000

**Comment by XxxSpiritvsxxX** [comment]
You need to install the latest SDK as well.
Not only RUNTIME, but also SDK 2.1.300 or Monitor will not work.

https://www.microsoft.com/net/download/windows

**Comment by Tomotoman** [comment]
Just installed the SDK. That didn't solve the problem

**Comment by LeeUmney** [comment]
I have installed both the runtime and SDK, uninstalled the previous versions, still get the 500 error.

**Comment by Slikkit** [comment]
I can confirm that I am also having the same issue. 

- I'm running Windows Server 2012 on VPS.
- 2.1 SDK (v2.1.300)
- 2.1 Runtime (v2.1.0) 
- PTM 2.0.2
- PT Ver:  2.0.10

**Comment by dwlahnum** [comment]
I can confirm that I took all of the advice given so far and still the same results. 

* I'm running Windows 10 on my desktop
* 2.1 SDK (v2.1.300)
* 2.1 Runtime (v2.1.0)
* PTM 2.0.2
* PT Ver: 2.0.10
* Was working fine on the same port with previous version 1.6.0, so no firewall blocks are taking place

**Comment by Legedric** [comment]
I am sorry but I am not able to reproduce this issue on Windows 10, Windows Server and on Mac OS. It has to be something local on your machine that caused this issue to occur but I do not know what it may be.

**Comment by dwlahnum** [comment]
I don't know what happened, but after multiple attempts to install older versions of PTMagic, uninstalling and installing multiple versions of dotnet. I finally ended up just deciding to install the newest version (2.0.3) with the dotnet recommended with that version and now all of the sudden the monitor is working as intended. Although the first thing I tried to do was upgrade to the 2.0.3 version as my first step, it still didn't work, so it must have been with installing/uninstalling all of the dotnet versions, possibly some of the uninstalls didn't totally clean things up until I did it a additional times or something. I hope that helps out the others who were having this same issue!

**Comment by d1rt** [comment]
The same issue


    I'm running Windows 10 on my desktop
    2.1 SDK (v2.1.300)
    2.1 Runtime (v2.1.0) (uninstalled all older versions)
    PTM 2.0.2
    PT Ver: 2.0.10
    Was working fine on the same port with previous version 1.6.0, so no firewall blocks are taking place
    No errors anywhere and still works like a champ, other than being able to view the monitor

**Comment by cash-money** [comment]
Same issue....

i am running PT Magic from Unbuntu..i have the following installed.  

i have made sure i am on the latest version, still have the problem...blank screen using firefox from a mac...

root@usve262206:~# dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   2.1.300
 Commit:    adab45bf0c

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  16.04
 OS Platform: Linux
 RID:         ubuntu.16.04-x64
 Base Path:   /usr/share/dotnet/sdk/2.1.300/

Host (useful for support):
  Version: 2.1.0
  Commit:  caa7b7e2ba

.NET Core SDKs installed:
  2.1.4 [/usr/share/dotnet/sdk]
  2.1.200 [/usr/share/dotnet/sdk]
  2.1.300 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.0 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.0 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.0.5 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.7 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.0 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
root@usve262206:~# 

Worked fine till the upgrade....

**Comment by flyboy1** [comment]
Same issue....
blank screen for Monitor url.... (HTTP/1.1 500 Internal Server Error)
PTMagic.dll runs without Problems...

Everything worked before the update:
1. uninstall old versions of dotnet
2. install new version (2.1) 
3. restart PTMagic and Monitor (PTMagic runs fine...)

root@ubuntu16-pt:/home/administrator# dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   2.1.300
 Commit:    adab45bf0c

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  16.04
 OS Platform: Linux
 RID:         ubuntu.16.04-x64
 Base Path:   /usr/share/dotnet/sdk/2.1.300/

Host (useful for support):
  Version: 2.1.0
  Commit:  caa7b7e2ba

.NET Core SDKs installed:
  2.1.300 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.0 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.0 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.0 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
root@ubuntu16-pt:/home/administrator#


root@ubuntu16-pt:/home/administrator# curl -vvvv http://0.0.0.0:5002
* Rebuilt URL to: http://0.0.0.0:5002/
*   Trying 0.0.0.0...
* Connected to 0.0.0.0 (127.0.0.1) port 5002 (#0)
> GET / HTTP/1.1
> Host: 0.0.0.0:5002
> User-Agent: curl/7.47.0
> Accept: */*
>
< HTTP/1.1 500 Internal Server Error
< Date: Fri, 08 Jun 2018 06:25:26 GMT
< Server: Kestrel
< Content-Length: 0
<
* Connection #0 to host 0.0.0.0 left intact
root@ubuntu16-pt:/home/administrator#

is there a way to make HTTP Error 500 visible with details?

I tried:
<httpErrors errorMode="Detailed" />
	<asp scriptErrorSentToBrowser="true"/>
in the web.config... without success

**Comment by LeeUmney** [comment]
I built a new Windows 10 VM, latest build, installing only Java for Profit trailer and the .Net requirements for PTMagic.
VM details:
4 CPU,
8GB RAM,
60GB HDD.
Profit Trailer runs perfectly, PT Magic.dll runs perfectly, Monitor returns 500 error.

Can you confirm the version of Windows you are testing this against to ensure it's not something that is in a recent update that is blocking this for some reason?

**Comment by michaelrgalloway** [comment]
I cannot get this working. Even wiped machine, started from scratch and still the monitor web server does not work. Same description as everyone else that has the issue.

**Comment by Softrunner** [comment]
At the moment I have the same problem that PTM not working anymore with the latest version?
I use a Ubuntu 16.4 Vps server and did also a complete clean install, but still no luck with installing and make it work again. Any suggestions? Maybe an update for the Linux docs?

Hopefully somebody can help me fix this problem(s)
I have the same issiue on my Raspberry Pi 3b+

**Comment by pkmelee337** [comment]
I have exactly the same problem. PT & PT Magic are working fine since I have updated, but the Monitor isn't reachable. (HTTP ERROR 500) I uninstalled previous dot net sdk versions, installed the latest sdk (with runtime), uninstalled the latest again and installed 2.1.0 (2.1.1 is the latest currently) and both are not working.

I tried to enable logging for the PT monitor, but this isn;t working either. The starting CMD says everthing is fine.

Has anyone a solution for this problem yet?
